src/types/Course/Section.ts

```diff
@@ -32,11 +33,75 @@ export const CourseQuery = extendType({
   type: 'Query',
   definition(t) {
     t.list.nonNull.field('courses', {
+      // type is a list of coursesections
       type: CourseSection,
-      description: 'Get a list of courses for a given term',
+      description: 'Get a list of courses for a given term and/or year',
       args: {
         term: arg({ type: Term }),
+        year: arg({ type: 'Int' }),
       },
+      resolve: async (_, args, ctx) => {
+        const { term, year } = args;
+        const { prisma } = ctx;
+        
+        let courses: Course[] = [];
+        let courseSections: any  = [];
+
+        // if term and year are provided, get courses for that term
+        if (term && year){
+          courses = await (prisma as PrismaClient).course.findMany({
+            where: {
+              term: term,
+              year: year,
+            },
+          });
+
+        }
+        else if (term){
+          courses = await (prisma as PrismaClient).course.findMany({
+            where: {
+              term: term,
+            },
+          });
+        }
+        else if (year){
+          courses = await (prisma as PrismaClient).course.findMany({
+            where: {
+              year: year,
+            },
+          });
+        }
+        else {
+          courses = await (prisma as PrismaClient).course.findMany();
+        }
+        console.log(courses);
+        // Get meeting times for each course
+        for (const course of courses) {
+          const meetingTimes = await (prisma as PrismaClient).meetingTime.findMany({
+            where: {
+              course: {
+                id: course.id,
+              },
+            }
+            });
+            courseSections.push({
+              CourseID: {
+                subject: course.subject,
+                code: course.code,
+                term: course.term,
+                year: course.year
+              },
+              hoursPerWeek: course.weeklyHours,
+              capacity: course.capacity,
+              professors: course.professorUsername,
+              startDate: course.startDate,
+              endDate: course.endDate,
+              meetingTimes: meetingTimes,
+            });
+        }
+
+        return courseSections;
```

Comment:
```suggestion
        return courses.map(({id, subject, code, term, year, weeklyHours, capacity, professorUsername, startDate, endDate}) => {
         const meetingTimes = await (prisma as PrismaClient).meetingTime.findMany({
            where: {
              course: {
                id,
              },
            }
            });
            return {
            courseId: {
             subject,
             code,
             term,
             year
            },
            hoursPerWeek: weeklyHours,
            capacity,
            professors: professorUsername,
            startDate,
            endDate,
            meetingTimes
            }
        });
```
again, syntax might be bad cause written in github

Reply:
This typescript stuff is blowing my mind